internal/cluster/cluster_test.go

```diff
@@ -2258,6 +2258,62 @@ var _ = Describe("GenerateAdditionalManifests", func() {
 	})
 })
 
+var _ = Describe("Deregister inactive clusters", func() {
+	var (
+		ctrl          *gomock.Controller
+		ctx           = context.Background()
+		db            *gorm.DB
+		state         API
+		c             common.Cluster
+		eventsHandler events.Handler
+		mockMetric    *metrics.MockAPI
+		dbName        = "permanently_delete_clusters"
+		mockS3Api     *s3wrapper.MockAPI
+	)
+
+	registerCluster := func() common.Cluster {
+		id := strfmt.UUID(uuid.New().String())
+		eventsHandler.AddEvent(ctx, id, &id, "", "", time.Now())
+		cl := common.Cluster{Cluster: models.Cluster{
+			ID:                 &id,
+			MonitoredOperators: []*models.MonitoredOperator{&common.TestDefaultConfig.MonitoredOperator},
+		}}
+		Expect(db.Create(&cl).Error).ShouldNot(HaveOccurred())
+		return cl
+	}
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		mockMetric = metrics.NewMockAPI(ctrl)
+		mockS3Api = s3wrapper.NewMockAPI(ctrl)
+		mockOperators := operators.NewMockAPI(ctrl)
+		db = common.PrepareTestDB(dbName)
+		eventsHandler = events.New(db, logrus.New())
+		dummy := &leader.DummyElector{}
+		state = NewManager(getDefaultConfig(), common.GetTestLog(), db, eventsHandler, nil, mockMetric, nil, dummy, mockOperators)
+		c = registerCluster()
+
+	})
+	It("Deregister inactive cluster testing_now", func() {
```

Comment:
please test a combination of clusters that needs to be deleted and those that are not deleted, we don't want to delete all the clusters in production because of a bug or something like this.